esphome/components/analog_threshold/analog_threshold_binary_sensor.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include "esphome/core/component.h"
+#include "esphome/components/binary_sensor/binary_sensor.h"
+#include "esphome/components/sensor/sensor.h"
+
+namespace esphome {
+namespace analog_threshold {
+
+class AnalogThresholdBinarySensor : public Component, public binary_sensor::BinarySensor {
+ public:
+  void dump_config() override;
+  void setup() override;
+
+  float get_setup_priority() const override { return setup_priority::HARDWARE; }
```

Comment:
This probably should not be HARDWARE as the sensor it is attaching to might not be setup yet
```suggestion
  float get_setup_priority() const override { return setup_priority::DATA; }
```

Reply:
I see you just changed this..haha

Reply:
Thought the same and has just pushed and update